### ISSUE TITLE
Multiple types for VarArgs incorrectly reported as "object"

### ISSUE THREAD

**Issue by groboclown** [python/mypy#7200]
MyPy version: 0.560
Python version: 3.6

I'm using a vararg setup that can take several types of data.  For example:

```python
from typing import TypeVar, List, Dict

ListInt = List[int]
DictInt = Dict[str, int]
Combo = TypeVar('Combo', ListInt, DictInt)

def arg_input(*vargs: Combo) -> None:
    pass
```

With this setup, these invocations are recognized correctly by MyPy:

```python
arg_input([1,2], [3,4])
arg_input({'a': 1}, {'b': 2})
```

But on mixed input:

```python
arg_input([1,2,3], { 'a': 1 })
```

mypy reports the error `Value of type variable "Combo" of "arg_input" cannot be "object"`

I would expect the inspection to allow the arguments to be any of the allowable types, rather than pinned to a single type.

**Comment by groboclown** [comment]
Looks like I'm using the wrong type.  It should be a `Union` instead of a `TypeVar`.

**Comment by JelleZijlstra** [comment]
Indeed. Also, please use a more recent version of mypy; we don't support old versions like 0.560 (which was released in December 2017).
